### PR TITLE
fix: improve sd-tag a11y

### LINKED ISSUE #1490: feat[dev]: ✨ implement A11y improvements to sd-tag
## Description
This task describes the Storybook implementation for the sd-tag component as part of the A11y improvements outlined in Epic https://github.com/solid-design-system/solid/issues/1465.

A11y improvements for sd-tag can be found here: [sd-tag Notes](https://www.notion.so/matuzo/sd-tag-10ef3f6e1bc080df951fcd49f55c5ffa).

--------------------------------------

## Comment
**Problem:** Selected tags communicate they're selected state purely visually.
**Solution:** 

- [x] Bind `aria-pressed="true|false"` to the `selected` to indicate that state semantically.
- [x] Add a new attribute `toggleable` to make sure the `aria-pressed` state is not used when being not a toggle (eg. a button or link).
- [x] Update the docs to explain/describe the new attributes and their purpose.
- [x] refactor the "Selected" story to show the exact behavior (select and de-select), add the "toggleable" attribute and add a hint to the story in order to make clear the 2 attributes are only working together
- [x] add code-only description to Figma

## Notes
- Disabled tags: See [@sd-button](https://www.notion.so/sd-button-104f3f6e1bc0800b9f72d2a046316ec3?pvs=24). Everything I wrote there about disabling form elements applies here as well.

## Related WCAG criteria
- https://www.w3.org/WAI/WCAG22/quickref/#name-role-value

## DoR
- [x] Item has been estimated by the team
- [x] Item dependencies have been identified and documented

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch

### PR DESCRIPTION
**Pull request by paulovareiro29**
## Description:

Closes #1490

Goal: Implement `aria-pressed` on the sd-tag and a `toggleable` attribute to control whenever the `aria-pressed` should show or not.

## Definition of Reviewable:

- [x] Documentation is created/updated
- [x] E2E tests (features, a11y, bug fixes) are created/updated

- [x] Stories (features, a11y) are created/updated
- [x] relevant tickets are linked

### CODE REVIEW

**Comment by paulovareiro29** [comment]
[sd-tag figma docs](https://www.figma.com/design/YDktJcseQIIQbsuCpoKS4V/Component-Docs?t=l4Fzt1sEmnSo7NN5-0) have been updated

**Review by smfonseca — APPROVED** [review]

**Review by Vahid1919 — DISMISSED** [review]

**Review by karlbaumhauer — DISMISSED** [review]

**Review by MartaPintoTeixeira — APPROVED** [review]

File: packages/docs/src/stories/components/tag.stories.ts [hunk 1]
```diff
@@ -51,12 +51,15 @@ export const Size = {
...
+ *
+ * __Accessibility hint__: Use the attribute `toggleable` to reflect the `selected` state on the `aria-pressed` attribute.
  */
 
 export const Selected = {
   render: () => html`
     <div class="flex gap-12">
-      <sd-tag selected>Selected</sd-tag>
+      <sd-tag selected toggleable>Selected</sd-tag>
+      <sd-tag toggleable>Unselected</sd-tag>
```
Comment (mariohamann) [inline packages/docs/src/stories/components/tag.stories.ts:62]: Could you please add some JS here to really make it toggle when you interact with it?

File: packages/docs/src/stories/components/tag.stories.ts [hunk 2]
```diff
@@ -51,13 +51,28 @@ export const Size = {
...
 export const Selected = {
   render: () => html`
-    <div class="flex gap-12">
-      <sd-tag selected>Selected</sd-tag>
+    <div id="tags-selected" class="flex gap-12">
+      <sd-tag selected toggleable>Selected</sd-tag>
+      <sd-tag toggleable>Unselected</sd-tag>
     </div>
+
+    <script>
```
Comment (smfonseca) [inline packages/docs/src/stories/components/tag.stories.ts:65]:
```suggestion
    <script type="module">
```

If you change to another component and back to the sd-tag, you get an error that `tag` and `handleToggle` is already been declared. Adding `type="module"` fixes the issue.

**Comment by paulovareiro29** [comment]
@MartaPintoTeixeira @coraliefeil could you update the Figma docs to resemble the code, on the Selected story for sd-tag?
The change is that I added an "Unselected" tag next to the existing "Selected" one.

**Comment by MartaPintoTeixeira** [comment]
> @MartaPintoTeixeira @coraliefeil could you update the Figma docs to resemble the code, on the Selected story for sd-tag?
> The change is that I added an "Unselected" tag next to the existing "Selected" one.

Usually in docs we try to make it as simple as possible to explain things. I believe here there is no need to show the Unselected option as that is the default one. I would suggest you do the same in code to be aligned.

**Review by auroraVasconcelos — APPROVED** [review]

**Review by MartaPintoTeixeira — APPROVED** [review]

**Review by mariohamann — APPROVED** [review]

**Pull request merged**